lib/react_client.dart

```diff
@@ -491,12 +491,16 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
         component.adapter = new JsComponent2Adapter(jsThis: jsThis);
         // Return the component so that the JS proxying component can store it,
         // avoiding an interceptor lookup.
+        component.init();
```

Comment:
I believe we'll want to call this after initializing jsThis/props/context below, otherwise those won't be able to be used in `init`